iceoryx_posh/test/moduletests/test_mepoo_shm_safe_unmanaged_chunk.cpp

```diff
@@ -0,0 +1,266 @@
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallDuplicateToSharedChunkOnSutPreviouslyCalledReleaseToSharedChunkResultsInEmptySharedChunk)
+{
+    auto sharedChunk = getChunkFromMemoryManager();
+
+    ShmSafeUnmanagedChunk sut(sharedChunk);
+    sut.releaseToSharedChunk();
+
+    EXPECT_FALSE(sut.duplicateToSharedChunk());
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallGetChunkHeaderOnNonConstDefaultConstructedSutResultsInNullptr)
+{
+    ShmSafeUnmanagedChunk sut;
+
+    EXPECT_EQ(sut.getChunkHeader(), nullptr);
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallGetChunkHeaderOnNonConstSutConstructedWithSharedChunkResultsInValidHeader)
+{
+    auto sharedChunk = getChunkFromMemoryManager();
+
+    ShmSafeUnmanagedChunk sut(sharedChunk);
+
+    auto chunkHeader = sut.getChunkHeader();
+    EXPECT_NE(chunkHeader, nullptr);
+    EXPECT_EQ(chunkHeader, sharedChunk.getChunkHeader());
+
+    sut.releaseToSharedChunk();
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallGetChunkHeaderOnConstDefaultConstructedSutResultsInNullptr)
+{
+    const ShmSafeUnmanagedChunk sut;
+
+    EXPECT_EQ(sut.getChunkHeader(), nullptr);
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallGetChunkHeaderOnConstSutConstructedWithSharedChunkResultsInValidHeader)
+{
+    auto sharedChunk = getChunkFromMemoryManager();
+
+    ShmSafeUnmanagedChunk sut(sharedChunk);
+
+    [&](const ShmSafeUnmanagedChunk& sut) {
+        auto chunkHeader = sut.getChunkHeader();
+        EXPECT_NE(chunkHeader, nullptr);
+        EXPECT_EQ(chunkHeader, sharedChunk.getChunkHeader());
+    } (sut);
+
+    sut.releaseToSharedChunk();
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallNonConstGetChunkHeaderResultsInNonConstChunkHeader)
+{
+    auto isNonConstReturn =
+        std::is_same<decltype(std::declval<ShmSafeUnmanagedChunk>().getChunkHeader()), ChunkHeader*>::value;
+    EXPECT_TRUE(isNonConstReturn);
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallConstGetChunkHeaderResultsInConstChunkHeader)
+{
+    auto isConstReturn =
+        std::is_same<decltype(std::declval<const ShmSafeUnmanagedChunk>().getChunkHeader()), const ChunkHeader*>::value;
+    EXPECT_TRUE(isConstReturn);
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallIsNotLogicalNullptrAndHasNoOtherOwnersOnDefaultConstructedResultsInFalse)
+{
+    ShmSafeUnmanagedChunk sut;
+
+    EXPECT_FALSE(sut.isNotLogicalNullptrAndHasNoOtherOwners());
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallIsNotLogicalNullptrAndHasNoOtherOwnersOnOnSutConstructedWithSharedChunkResultsInTrue)
+{
+    ShmSafeUnmanagedChunk sut(getChunkFromMemoryManager());
+
+    EXPECT_TRUE(sut.isNotLogicalNullptrAndHasNoOtherOwners());
+
+    sut.releaseToSharedChunk();
+}
+
+TEST_F(ShmSafeUnmanagedChunk_test, CallIsNotLogicalNullptrAndHasNoOtherOwnersOnOnSutConstructedWithSharedChunkAndOtherOwnerResultsInFalse)
+{
+    auto sharedChunk = getChunkFromMemoryManager();
+
+    ShmSafeUnmanagedChunk sut(sharedChunk);
+
+    EXPECT_FALSE(sut.isNotLogicalNullptrAndHasNoOtherOwners());
```

Comment:
I would add two more tests here.

1. let the `sharedChunk` from this test go out of scope and then `isNotLogicalNullptrAndHasNoOtherOwners` should be true again.
2. call `isNotLogicalNullptrAndHasNoOtherOwners` after you released a valid shared chunk with `releaseToSharedChunk`